### DISCUSSION TITLE
$NOW is not working in deep filter rule. Any thoughts?

### DISCUSSION THREAD

**Discussion by pchebro** [directus/directus#9319] [Archive]
Hi,
I am using below query for filters (deep) but have an issue with date condition with $NOW. The requirement is to retrieve items greater than or equal to current date. If items expiration date is greater than today's date then items are returning fine. But equal condition is not working with _gte. Please let me know if I am missing anything.

```json
{
  "home_page_news": {
    "_sort": "-display_date",
    "_limit": 5,
    "_filter": {
      "_and": [
        {
          "status": {
            "_in": [
              "published"
            ]
          }
        },
        {
          "_and": [
            {
              "effective_date": {
                "_lte": "$NOW"
              }
            },
            {
              "effective_date": {
                "_nnull": ""
              }
            }
          ]
        },
        {
          "_or": [
            {
              "expiration_date": {
                "_gte": "$NOW"
              }
            },
            {
              "expiration_date": {
                "_null": ""
              }
            }
          ]
        }
      ]
    }
  }
}

**Comment by rijkvanzanten** [comment]
This looks good to me. I'd use true/false for null / nnull personally.

What's the issue you're experiencing? Have you tried using the $NOW exclusively without all the other and/or groups?

**Reply by pchebro** [reply]
Thanks for your response. If there is an item with expiration date equal to current date, I expect the item with current date as expiration date plus greater than current date should return. But it's not returning items that are equal to current date. So basically return all items from collection where expiration date >= current date. Hope this clarifies your question. I suspect the issue is around date comparison.

I am having issue around below filter:

```json
{
    "expiration_date": {
        "_gte": "$NOW"
    }
}
```

**Reply by rijkvanzanten** [reply]
> If there is an item with expiration date equal to current date,

"$NOW" equals the current date and time, so if you compare that to a date, the time might be different.

What type of field are you using for "expiration_date"?

**Reply by pchebro** [reply]
expiration_date is a date field. I thought so but 'today' is not working though. Is that supported only for lower version of directus?

**Reply by pchebro** [reply]
Is there a global query parameter I can use for current date without timestamp?

**Reply by rijkvanzanten** [reply]
> expiration_date is a date field. I thought so but 'today' is not working though. Is that supported only for lower version of directus?

There is no `today` flag, that's correct. I agree that "$NOW" should exclusively use the current _date_ excluding the time 👍🏻
